components/kyma-environment-broker/internal/storage/driver/memory/instance.go

```diff
@@ -160,34 +158,25 @@ func (s *Instance) GetInstanceStats() (internal.InstanceStats, error) {
 	return internal.InstanceStats{}, fmt.Errorf("not implemented")
 }
 
-func (s *Instance) List(limit int, cursor string) ([]internal.Instance, *pagination.Page, int, error) {
+func (s *Instance) List(pageSize int, page int) ([]internal.Instance, int, int, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	var toReturn []internal.Instance
-
-	offset, err := pagination.DecodeOffsetCursor(cursor)
-	if err != nil {
-		return nil, nil, 0, err
+	var offset int
+	if page < 2 {
+		offset = 0
+	} else {
+		offset = page*pageSize - 1
```

Comment:
maybe an overhead but we have similar logic in the `ConvertPageLimitAndOrderedColumnToSQL` function, we could export it to e.g. separate function